### ISSUE TITLE
Inconsistent performance of SVTR and its ONNX model

### ISSUE THREAD

**Issue by thsno02** [PaddlePaddle/PaddleOCR#8631]
Related issues, the ONNX SVTR model yielded NULL result:

+ https://github.com/PaddlePaddle/PaddleOCR/issues/7821,
+ https://github.com/PaddlePaddle/PaddleOCR/issues/7634.

In my case, the ONNX SVTR model predicts the wrong results compared to the inference model, such as the inference model yields `2020-10-10 12:12:12` while the ONNX predicts `020-213:10`.

I have updated the paddle2onnx and onnxruntime packages to the latest version and used the following commands to convert and infer.

```
# convert inference model
python3 ./PaddleOCR-2.6/tools/export_model.py \
        -c .../config.yml \
        -o Global.pretrained_model=..../best_accuracy.pdparams
# convert ONNX model
paddle2onnx --model_dir ... \
            --model_filename ... \
            --params_filename ... \
            --save_file ... \
            --enable_dev_version True \
            --opset 16 \
            --enable_onnx_checker True
# optimze ONNX model
# h, w are the same as the image shape in the config.yml respectively
python -m paddle2onnx.optimize --input_model ... \
                               --outpu_model ... \
                               --input_shape_dict "{'x':[1,3,h,w]}"
# infer
python3 ./PaddleOCR-2.6/tools/infer/predict_rec.py \
        --use_onnx=True \
        --rec_model_dir='.../model.onnx' \
        --image_dir='.../test.jpg' \
        --rec_algorithm='SVTR' \
        --rec_image_shape='3,h,w' \
        --rec_char_dict_path='.../ocr_dict.txt'
```

The pre-trained model (SVTR Tiny ch) doesn't have this problem, though the inference result is slightly different from the ONNX, such as the inference model predicted 4 to 9 and ONNX didn't. I suspect that some parameters were lost in SVTR during the ONNX conversion.

Also, upgrading the TensorRT from 8.4 to 8.5 doesn't help.

For the model:
+ removed STN_ON block in architecture part;
+ changed some parameters in the backbone;
+ set dropout value to 0.5 in `rec_svtrnet.py`.

**Comment by Topdu** [comment]
SVTR_ch runs fine, as you say, but after modifying the parameter configuration, it runs incorrectly. If it is convenient to provide specific details of the modification it can help us to analyse it.

**Comment by Topdu** [comment]
`out_char_num` is appropriately set to w//4 or w//8. And for w=360, `out_char_num` is most appropriately set to w//8=45.

**Comment by thsno02** [comment]
It works.
